### PR TITLE
Create SRS SST layers with generic timestep schema

### PR DESCRIPTION
**Pull request by lbesnard**
To be tested on RC

### CODE REVIEW

File: wps/srs_ghrsst.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    <variable name="time"/>
+    <variable name="lat"/>
+    <variable name="lon"/>
+    <variable name="dt_analysis" type="Float"/>
+    <variable name="l2p_flags"/>
+    <variable name="quality_level"/>
+    <variable name="satellite_zenith_angle" type="Float"/>
+
+    <variable name="sea_surface_temperature" type="Float">
+      <attribute name="_FillValue" type="Double" value="9.96920996838687e+36"/>
```
Comment (jonescc) [inline wps/srs_ghrsst.xml]: Should probably set this to the same type as the variable - i.e. Float - ideally you should be able to leave it out and it be set correctly but looking at the code it looks mandatory - I'll raise an issue to allow this to be defaulted to the variable type if left out

**Comment by lbesnard** [comment]
Thanks @jonescc, I copied and pasted a bit blindly the template. I modified it and add the good variable types. Let me know if these will be handled

File: wps/srs_ghrsst.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    <attribute name="time_coverage_start" type="String" value="${TIME_START}"/>
+    <attribute name="time_coverage_end" type="String" value="${TIME_END}"/>
+    <attribute name="title" match=".*" type="String" value="${0}, ${TIME_START}, ${TIME_END}"/>
+    <attribute name="southernmost_latitude" type="Double" value="${LAT_MIN}"/>
+    <attribute name="northernmost_latitude" type="Double" value="${LAT_MAX}"/>
+    <attribute name="westernmost_longitude" type="Double" value="${LON_MIN}"/>
+    <attribute name="easternmost_longitude" type="Double" value="${LON_MAX}"/>
+  </attributes>
+  <variables>
+    <variable name="time" type="int"/>
```
Comment (jonescc) [inline wps/srs_ghrsst.xml]: You only need to specify the type here if you want to change it from the source files.

File: wps/srs_ghrsst.xml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <variable name="lat" type="float"/>
+    <variable name="lon" type="float"/>
+    <variable name="dt_analysis" type="byte"/>
+    <variable name="l2p_flags" type="short"/>
+    <variable name="quality_level" type="byte"/>
+    <variable name="satellite_zenith_angle" type="byte"/>
+    <variable name="sea_surface_temperature" type="short"/>
```
Comment (jonescc) [inline wps/srs_ghrsst.xml]: By default sea_surface_temperature would be unpacked to double (type of scale/offset attributes).  Here I think we were wanting to unpack it to 'Float' so save space.   So I'd change the type to 'Float'

File: wps/srs_ghrsst.xml (hunk 2 above)
Comment (jonescc) [inline wps/srs_ghrsst.xml]: I'd remove it from all variables here except for sea_surface_temperature as noted below.

**Comment by lbesnard** [comment]
thanks @jonescc, if that looks good to you, i'll squash those commits

File: wps/srs_ghrsst.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    <attribute name="southernmost_latitude" type="Double" value="${LAT_MIN}"/>
+    <attribute name="northernmost_latitude" type="Double" value="${LAT_MAX}"/>
+    <attribute name="westernmost_longitude" type="Double" value="${LON_MIN}"/>
+    <attribute name="easternmost_longitude" type="Double" value="${LON_MAX}"/>
+  </attributes>
+  <variables>
+    <variable name="time"/>
+    <variable name="lat"/>
+    <variable name="lon"/>
+    <variable name="dt_analysis"/>
```
Comment (jonescc) [inline wps/srs_ghrsst.xml]: Sorry- not familiar with the dataset. dt_analysis, satellite_zenith_angle, sses_bias, sses_count, sses_standard_deviation and sst_dtime look like they would also be unpacked to double and we want to use Float to save space.

File: wps/gogoduck.xml [hunk 5]
```diff
@@ -20,4 +20,9 @@
     <timeStart>time_coverage_start</timeStart>
     <timeEnd>time_coverage_end</timeEnd>
   </globalAttributes>
+  <!-- aggregation templates -->
+    <templates>
+      <template match="imos:srs_ghrsst.*">srs_ghrsst</template>
+      <template match=".*">default</template>
```
Comment (jonescc) [inline wps/gogoduck.xml]: Just noticed this line.  We would need to add a default.xml or remove this line so existing config is used for other datasets.

Comment (jonescc) [inline wps/gogoduck.xml]: An example default.xml which would do the same as gogoduck currently does by default (but correcting the type of extent attributes) would be:

```
<?xml version="1.0" encoding="UTF-8"?>
<template>
  <attributes>
    <attribute name="time_coverage_start" type="String" value="${TIME_START}"/>
    <attribute name="time_coverage_end" type="String" value="${TIME_END}"/>
    <attribute name="title" match=".*" type="String" value="${0}, ${TIME_START}, ${TIME_END}"/>
    <attribute name="geospatial_lat_min" type="Double" value="${LAT_MIN}"/>
    <attribute name="geospatial_lat_max" type="Double" value="${LAT_MAX}"/>
    <attribute name="geospatial_lon_min" type="Double" value="${LON_MIN}"/>
    <attribute name="geospatial_lon_max" type="Double" value="${LON_MAX}"/>
  </attributes>
</template>
```

**Comment by jonescc** [comment]
OK, aggregation templates look good.  We should be able to remove any of the old gogoduck.xml files we used to used to control this if you are happy with the result.

**Comment by jonescc** [comment]
Note that the default.xml file will be applied to non-srs datasets so need to check they are working as expected (ACORN, GSLA, CARS etc).  This moves us from the old way of configuring global attributes to the new for all datasets which is good. 

@ggalibert for info - the required template changes to change min/max lat and lon attributes to double referred to in https://github.com/aodn/geoserver-build/issues/228 are made in this PR.

**Pull request merged**